x/ibc-hooks/contracts/crosschain-swaps/src/contract.rs

```diff
@@ -0,0 +1,115 @@
+#[cfg(not(feature = "imported"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Reply, Response, StdResult,
+};
+use cw2::set_contract_version;
+
+use crate::consts::{FORWARD_REPLY_ID, SWAP_REPLY_ID};
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg, SudoMsg};
+use crate::state::{Config, CHANNEL_MAP, CONFIG, RECOVERY_STATES};
+use crate::{execute, sudo};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:crosschain-swaps";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Handling contract instantiation
+#[cfg_attr(not(feature = "imported"), entry_point)]
```

Comment:
Nice, this looks more straightforward than "library" and not be confused with `cargo build --lib`

Reply:
Agreed! have an open issue for it here: https://github.com/CosmWasm/cw-template/issues/140